content/en/blog/2021-08-15-seccomp-default.md

```diff
@@ -0,0 +1,259 @@
+---
+layout: blog
+title: "Hardening Kubernetes security by enabling seccomp by default"
+date: 2021-08-15
+slug: seccomp-default
+---
+
+**Author:** Sascha Grunert, Red Hat
+
+_Editor's note: Sascha is member of [SIG Release][sigr] and is working on many
+other different container related topics. Feel free to reach him out on Twitter
+[@saschagrunert][twitter] or via the [Kubernetes slack @sascha][slack]._
+
+[sigr]: https://github.com/kubernetes/sig-release
+[twitter]: https://twitter.com/saschagrunert
+[slack]: https://kubernetes.slack.com/team/U53SUDBD4
+
+---
+
+Applying seccomp profiles to containerized workloads is one of the key tasks
+when it comes to enhancing the security of the application deployment.
+Developers, site reliability engineers and infrastructure administrators have to
+work hand in hand to create, distribute and maintain the profiles over the
+applications life-cycle.
+
+Kubernetes introduced dedicated [API fields][api] in the `SecurityContext` with
+the [graduation of seccomp to General Availability (GA)][ga] in v1.19.0. This
+enhancement allowed an easier way to specify if the whole pod or a specific
+container should run as:
+
+[api]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#seccompprofile-v1-core
+[ga]: https://github.com/kubernetes/enhancements/blob/cccc6fa/keps/sig-node/135-seccomp/README.md#seccomp-to-ga
+
+- `Unconfined`: seccomp will not be enabled
+- `RuntimeDefault`: the container runtimes default profile will be used
+- `Localhost`: a node local profile will be applied, which is being referenced
+  by a relative path to the seccomp profile root (`<kubelet-root-dir>/seccomp`)
+  of the kubelet
+
+With the graduation of seccomp, nothing has changed from an overall security
+perspective, because `Unconfined` is still the default. This is totally fine if
+we look from the upgrade path and backwards compatibility perspective of
+Kubernetes releases. But it also means that it is more likely that a workload
+runs without seccomp at all, which should be fixed in the long term.
+
+## `SeccompDefault` to the rescue
+
+Kubernetes v1.22.0 introduces a new kubelet [feature gate][gate]
+`SeccompDefault`, which has been added in `alpha` state as every other new
+feature. This means that it is disabled by default and can be enabled manually
+for every single Kubernetes node.
+
+[gate]: https://k8s.io/docs/reference/command-line-tools-reference/feature-gates
```

Comment:
We prefer site-relative hyperlinks, ie `/docs/reference/command-line-tools-reference/feature-gates/`

Reply:
Fixed :+1: